app/src/main/java/com/antonioleiva/bandhookkotlin/data/mapper/ArtistMapper.kt

```diff
@@ -24,20 +24,30 @@ import com.antonioleiva.bandhookkotlin.domain.entity.Artist
 class ArtistMapper {
 
     fun transform(artists: List<LastFmArtist>): List<Artist> {
-        return artists.filter { artistHasQualityInfo(it) }.map { transform(it) }
+        return artists.filter { artistHasQualityInfo(it) }.mapNotNull { transform(it) }
     }
 
-    fun transform(artist: LastFmArtist): Artist = Artist(
-            artist.mbid,
-            artist.name,
-            getImage(artist.images),
-            artist.bio?.content)
+    fun transform(artist: LastFmArtist): Artist? {
+        if (artist.mbid != null) {
```

Comment:
What do you think if we use the power of kotlin a little bit here? Something like:

```
fun transform(artist: LastFmArtist) = artist.mbid?.let {
    return Artist(
                    artist.mbid,
                    artist.name,
                    getImageUrl(artist.images),
                    artist.bio?.content)
}
```

I'm not totally convinced with this anyway, as it's a bit edgy case of `let` use, but let me know what you think.